File: golang/sorting/bubble_sort.go
package main

import "fmt"

func Swap(a, b int) (int, int) {
	return b, a
}

func BubbleSort(arr []int, size int) []int {
	for i := 0; i < size; i++ {
		for j := 0; j < size-i-1; j++ {
			if arr[j] > arr[j+1] {
				arr[j], arr[j+1] = Swap(arr[j], arr[j+1])
			}
		}
	}
	return arr
}

func main() {
	var arr []int = []int{98, 4, 2, 7, 5, 10, 33, 1}
	arr = BubbleSort(arr, len(arr))
	fmt.Println(arr)
}

File: python/sorting/bubble_sort.py
def bubble_sort(arr):
    # best case - O(n), avg, worst case - O(n^2)
    for i in range(len(arr)):
        for j in range(len(arr)-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
    return arr

if __name__ == "__main__":
    arr = [98, 4, 2, 7, 5, 10, 33, 1]
    str_arr = ['z', 'y', 'x', 'w', 'v', 'u']
    # compare the adjacent value
    assert bubble_sort(arr) == [1, 2, 4, 5, 7, 10, 33, 98]
    assert bubble_sort(str_arr) == ['u', 'v', 'w', 'x', 'y', 'z']
